### ISSUE TITLE
Genomic_converter error : haplotypes to vcf

### ISSUE THREAD

**Issue by faridasamadzada** [thierrygosselin/radiator#54]
Hi Thierry,
I'm trying to use genome_converter in radiator to convert haplotypes file generated by Stacks v1.3 to vcf and I get the following error:

Error in dplyr::if_else(POLYMORPHISM == 0, stringi::stri_join(GT_VCF_NUC, : object 'POLYMORPHISM' not found.

The code I used:

`genomic_converter(data = batch_1.haplotypes.tsv, strata = map.txt, output = c("vcf"), verbose = TRUE)`

Traceback:

13.
dplyr::if_else(POLYMORPHISM == 0, stringi::stri_join(GT_VCF_NUC, "/", GT_VCF_NUC), GT_VCF_NUC, missing = "./.")
12.
mutate_impl(.data, dots, caller_env())
11.
mutate.tbl_df(., GT_VCF_NUC = dplyr::if_else(POLYMORPHISM == 0, stringi::stri_join(GT_VCF_NUC, "/", GT_VCF_NUC), GT_VCF_NUC, missing = "./."), GT_VCF_NUC = dplyr::if_else(stringi::stri_detect_fixed(GT_VCF_NUC, "N"), "./.", GT_VCF_NUC))
10.
dplyr::mutate(., GT_VCF_NUC = dplyr::if_else(POLYMORPHISM == 0, stringi::stri_join(GT_VCF_NUC, "/", GT_VCF_NUC), GT_VCF_NUC, missing = "./."), GT_VCF_NUC = dplyr::if_else(stringi::stri_detect_fixed(GT_VCF_NUC, "N"), "./.", GT_VCF_NUC))
9.
function_list[[i]](value)
8.
freduce(value, `_function_list`)
7.
`_fseq`(`_lhs`)
6.
eval(quote(`_fseq`(`_lhs`)), env, env)
5.
eval(quote(`_fseq`(`_lhs`)), env, env)
4.
withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
3.
input %>% dplyr::mutate(GT_VCF_NUC = dplyr::if_else(POLYMORPHISM == 0, stringi::stri_join(GT_VCF_NUC, "/", GT_VCF_NUC), GT_VCF_NUC, missing = "./."), GT_VCF_NUC = dplyr::if_else(stringi::stri_detect_fixed(GT_VCF_NUC, "N"), "./.", GT_VCF_NUC)) %>% dplyr::select(-POLYMORPHISM)
2.
tidy_genomic_data(data = data, strata = strata, filename = filename, parallel.core = parallel.core, whitelist.markers = whitelist.markers, blacklist.id = blacklist.id, vcf.metadata = vcf.metadata, vcf.stats = vcf.stats, keep.allele.names = keep.allele.names

I would like to know whether the error is due to the input file. (I'm using an older version of Stacks). 
I checked the file with detect_genomic_format and radiator recognizes the file as haplo.file.

Thanks in advance for your help!
Best,
Farida

**Comment by thierrygosselin** [comment]
Hi Farida,

I think I know where the problem comes from,
could you run for me the same thing but using:

```
test1 <-  radiator::tidy_genomic_data(data = "batch_1.haplotypes.tsv", strata = "map.txt", verbose = TRUE)
```

**Comment by faridasamadzada** [comment]
Looks like it gives me the same error:

Error in dplyr::if_else(POLYMORPHISM == 0, stringi::stri_join(GT_VCF_NUC, : object 'POLYMORPHISM' not found
12.
dplyr::if_else(POLYMORPHISM == 0, stringi::stri_join(GT_VCF_NUC, "/", GT_VCF_NUC), GT_VCF_NUC, missing = "./.")
11.
mutate_impl(.data, dots, caller_env())
10.
mutate.tbl_df(., GT_VCF_NUC = dplyr::if_else(POLYMORPHISM == 0, stringi::stri_join(GT_VCF_NUC, "/", GT_VCF_NUC), GT_VCF_NUC, missing = "./."), GT_VCF_NUC = dplyr::if_else(stringi::stri_detect_fixed(GT_VCF_NUC, "N"), "./.", GT_VCF_NUC))
9.
dplyr::mutate(., GT_VCF_NUC = dplyr::if_else(POLYMORPHISM == 0, stringi::stri_join(GT_VCF_NUC, "/", GT_VCF_NUC), GT_VCF_NUC, missing = "./."), GT_VCF_NUC = dplyr::if_else(stringi::stri_detect_fixed(GT_VCF_NUC, "N"), "./.", GT_VCF_NUC))
8.
function_list[[i]](value)
7.
freduce(value, `_function_list`)
6.
`_fseq`(`_lhs`)
5.
eval(quote(`_fseq`(`_lhs`)), env, env)
4.
eval(quote(`_fseq`(`_lhs`)), env, env)
3.
withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
2.
input %>% dplyr::mutate(GT_VCF_NUC = dplyr::if_else(POLYMORPHISM == 0, stringi::stri_join(GT_VCF_NUC, "/", GT_VCF_NUC), GT_VCF_NUC, missing = "./."), GT_VCF_NUC = dplyr::if_else(stringi::stri_detect_fixed(GT_VCF_NUC, "N"), "./.", GT_VCF_NUC)) %>% dplyr::select(-POLYMORPHISM)
1.
radiator::tidy_genomic_data(data = "batch_1.haplotypes.tsv", strata = "map.txt", verbose = TRUE)

**Comment by thierrygosselin** [comment]
Hi Farida,

Could you send by email the required files so that I can reproduce the error.

And 2 quick questions:
1. Is there a reason you're not using a newer version of stacks ? Lots of bug fixed since v1.3...
2. Do you have the VCF file associated with the haplotypes file ?

Cheers
Thierry

**Comment by thierrygosselin** [comment]
Hi Farida,

The problem with the haplotype file you have was modified and radiator could not parse it correctly. Usually the first column, the locus in the catalog is: `# Catalog ID`, `Catalog ID`, `# Catalog Locus ID`. It's the first time I see `Catalog.ID`.

The new commit on GitHub can now take care of this.

However, the current implementation of `genomic_converter` will throw an error to convert to VCF:
` Error: vcf output is currently implemented for biallelic data only`, but it will work with other format like `genepop`.

Try this with the data you sent:
```
test <- radiator::genomic_converter(
data = "batch_1.haplotypes.txt", 
strata = "AluAndSetEditmap.txt", 
output = c("genepop", "vcf"), 
verbose = TRUE
)
```
